### ISSUE TITLE
tNotes crashes on leading zeroes in verse links

### ISSUE THREAD

**Issue by PhotoNomad0** [unfoldingWord/translationCore#7434]
### User Story

tNotes crashes on leading zeroes in verse links that are embedded in the notes.

**_Field reported issue._**

**Reproducing the Problem:**
Can reproduce in this project: 
[gu_gst_jhn_book.zip](https://github.com/unfoldingWord/translationCore/files/9650174/gu_gst_jhn_book.zip)

Here is the log file: 
[Sat-Sep-17-2022.log](https://github.com/unfoldingWord/translationCore/files/9650180/Sat-Sep-17-2022.log)

You can also reproduce the problem in any John project. Open tN tool with en/Door43-Catalog as gateway language and select group `Assumed Knowledge and Implicit Information`. Then select 20:8 and you will see the crash. Doing tN checks is now a ticking time bomb, because it does not recover gracefully. And it will crash the next time you start tN. Because it remembers the last verse you selected and tries to re-open it each time you start tN.

**Work-Around:**
There is a "hacky" work-around. Before launching tN, you can change the gateway language to hi (which doesn't have the verse link in the note). Then launch tN and move to the next check. Then you can reopen tNotes with english as the GL. I know it is a messy work-around, but at least they would be able to do some work until a fix is out.

### Features / Specifications

- [ ]
- [ ]
- [ ]

### Definition of Done

- [ ]
- [ ]
- [ ]

**Comment by PhotoNomad0** [comment]
@elsylambert This is fixed in `translationCore 3.4.0 (a716bfb)`

You can test with the example above or any book for John with en/Door43-Catalog as the gateway of tNotes.  Click on the selection given above, and should not see crash.  A better test would be this check:

![Screen Shot 2022-09-27 at 1 54 20 PM](https://user-images.githubusercontent.com/14238574/192601801-f98a5748-3cf7-476e-b923-d415582b57ca.png)

It has a leading zero for both the chapter and verse (from https://git.door43.org/Door43-Catalog/en_tn/raw/branch/master/en_tn_44-JHN.tsv):

```
JHN	1	13	ygxb	figs-metaphor	οἳ οὐκ ἐξ αἱμάτων…ἐγεννήθησαν	1		John uses **born** figuratively to describe God changing a person from being spiritually dead to being spiritually alive when they believe in Jesus. John recorded Jesus referring to this change as being “born again” in [3:3](../03/03.md). (See: [[rc://en/tw/dict/bible/kt/bornagain]]) If your readers would misunderstand this, you could use an expression that indicates spiritual rebirth rather than physical birth. Alternate translation: “These were born spiritually,
```

You can also hover over the **3:3** to make sure it was able to read and show the verse text.

**Comment by elsylambert** [comment]
Not seeing the crash in translationCore 3.4.0 (a716bfb). Tested in John and PHP projects. Verse links shows verses on-hover.
